packages/talos_messenger_core/src/core.rs

```diff
@@ -0,0 +1,46 @@
+use ahash::HashMap;
+use async_trait::async_trait;
+use serde::{Deserialize, Serialize};
+use serde_json::Value;
+use strum::{Display, EnumIter, EnumString};
+
+use crate::errors::MessengerServiceResult;
+
+#[derive(Debug, Display, Serialize, Deserialize, EnumString, EnumIter, Clone, Eq, PartialEq)]
+pub enum CommitActionType {
+    #[strum(serialize = "publish")]
+    Publish,
+}
+
+#[derive(Debug, Display, Serialize, Deserialize, EnumString, Clone, Eq, PartialEq)]
+pub enum PublishActionType {
+    #[strum(serialize = "kafka")]
+    Kafka,
+}
+
+#[async_trait]
+pub trait MessengerPublisher {
+    type Payload;
+    type AdditionalData;
+    fn get_publish_type(&self) -> PublishActionType;
+    async fn send(&self, version: u64, payload: Self::Payload, additional_data: Self::AdditionalData) -> ();
+}
+
+/// Trait to be implemented by all services.
+#[async_trait]
+pub trait MessengerSystemService {
+    async fn start(&self) -> MessengerServiceResult;
+    async fn run(&mut self) -> MessengerServiceResult;
+    async fn stop(&self) -> MessengerServiceResult;
+}
+
+#[derive(Debug)]
+pub struct MessengerCommitActions {
+    pub version: u64,
+    pub commit_actions: HashMap<String, Value>,
+}
+
+pub enum MessengerChannelFeedback {
+    Error(u64, String),
```

Comment:
Why not to make simple structs or named variables. Tuple style allows only to show types but names and purposes are very not clear :)

Reply:
Agree, just the lazy me to blame :) I will update this

Reply:
Sorry, I missed this one while updating the review comments changes. Will update it in the next PR